pdns/validate.cc

```diff
@@ -512,6 +512,16 @@ dState getDenial(const cspmap_t &validrrsets, const DNSName& qname, const uint16
            continue;
         }
 
+        /* The NSEC is either a delegation one, from the parent zone, and
+         * must have the NS bit set but not the SOA one, or a regular NSEC
+         * either at apex (signer == owner) or with the SOA or NS bits clear.
+         */
+        const bool notApex = signer.countLabels() < owner.countLabels();
```

Comment:
Isn't this test equivalent to `signer != owner` as we know `owner.isPartOf(signer)`, as established by the `if (... || !owner.isPartOf(signer)) continue` test above?

Same for NSEC3 case.